### ISSUE TITLE
Document DevTools functionality and usage

### ISSUE THREAD

**Issue by parlough** [dart-lang/site-www#4352]
### What information is missing?

We currently document what DevTools is and how to start using it, but don't document how to use it to debug, profile, or analyze code. The Flutter docs have documentation on using many of its views but that's a different site and also contains a lot of Flutter specific mentions. 

### How would you like us to fix this problem?

We should have more extensive documentation on the various DevTools views that is applicable to all types of Dart applications.

**Comment by johnpryan** [comment]
We might want to make sure we don't duplicate docs between dart.dev and flutter.dev cc: @kenzieschmoll

**Comment by kenzieschmoll** [comment]
There are some DevTools tools that apply to both standalone dart apps and to flutter apps, though a lot of the content on flutter.dev assumes a flutter app is connected. On many pages (like the performance page for example) there will be a smaller subset of features that are enabled for Dart CLI apps, so it may be worth documenting those workflows on dart.dev. Can we share code between the two websites? for the pages that are the same for standalone Dart and Flutter, it could be useful to share the code and display the page on both sites.

**Comment by johnpryan** [comment]
I don't think we have a way to share docs between the two sites. We _might_ be able to use site-shared, but this isn't something we are doing right now as far as I know.

**Comment by atsansone** [comment]
@johnpryan : That's exactly what we could use `site-shared` for. I'd need to take a look at the structure to better understand any issues with doing so.
